src/Raven.Server/Documents/Sharding/Handlers/Processors/Streaming/ShardedStreamingHandlerProcessorForGetStreamQuery.cs

```diff
@@ -0,0 +1,237 @@
+using Raven.Client.Extensions;
+using Raven.Client.Http;
+using Raven.Server.Documents.Commands.Streaming;
+using Raven.Server.Documents.Handlers;
+using Raven.Server.Documents.Handlers.Processors.Streaming;
+using Raven.Server.Documents.Queries;
+using Raven.Server.Documents.Replication.Senders;
+using Raven.Server.Documents.Sharding.Operations;
+using Raven.Server.Documents.Sharding.Queries;
+using Raven.Server.NotificationCenter;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Sparrow.Json;
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents.Sharding.Handlers.Processors.Streaming
+{
+    internal class ShardedStreamingHandlerProcessorForGetStreamQuery : AbstractStreamingHandlerProcessorForGetStreamQuery<ShardedDatabaseRequestHandler, TransactionOperationContext>
+    {
+        public ShardedStreamingHandlerProcessorForGetStreamQuery([NotNull] ShardedDatabaseRequestHandler requestHandler, HttpMethod method) : base(requestHandler, method)
+        {
+        }
+        
+        protected override RequestTimeTracker GetTimeTracker()
+        {
+            return new RequestTimeTracker(HttpContext, Logger, RequestHandler.DatabaseContext.NotificationCenter, RequestHandler.DatabaseContext.Configuration, "StreamQuery", doPerformanceHintIfTooLong: false);
+        }
+
+        protected override async ValueTask<BlittableJsonReaderObject> GetDocumentData(TransactionOperationContext context, string fromDocument)
+        {
+            var shard = RequestHandler.DatabaseContext.GetShardNumber(context, fromDocument);
+            
+            var docs = await RequestHandler.ShardExecutor.ExecuteSingleShardAsync(context,
+                new GetDocumentsCommand(new string[] { fromDocument }, null, metadataOnly: false), shard);
+            return (BlittableJsonReaderObject)docs.Results[0];
+        }
+
+        protected override IDisposable AllocateContext(out TransactionOperationContext context)
+        {
+            return ContextPool.AllocateOperationContext(out context);
+        }
+
+        protected override QueryMetadataCache GetQueryMetadataCache()
+        {
+            return RequestHandler.DatabaseContext.QueryMetadataCache;
+        }
+
+        protected override IStreamQueryResultWriter<BlittableJsonReaderObject> GetBlittableQueryResultWriter(string format, bool isDebug, JsonOperationContext context, HttpResponse response, Stream responseBodyStream, bool fromSharded,
+            string[] propertiesArray, string fileNamePrefix = null)
+        {
+            if (IsCsvFormat(format))
+            {
+                //does not write query stats to stream
+                return new StreamCsvBlittableQueryResultWriter(response, responseBodyStream, propertiesArray, fileNamePrefix);
+            }
+            
+            if (isDebug)
+            {
+                ThrowUnsupportedException($"You have selected \"{format}\" file format, which is not supported.");
```

Comment:
It looks the exception message is wrong since it talks about file format, not about debug mode

Reply:
the debug mode is checked outside in the abstract. The `isDebug` is to know which "path" I came from